src/test/regress/citus_tests/common.py

```diff
@@ -572,6 +578,21 @@ def sql(self, query, params=None, **kwargs):
         with self.cur(**kwargs) as cur:
             cur.execute(query, params=params)
 
+    def sql_row(self, query, params=None, allow_empty_result=False, **kwargs):
+        """Run an SQL query that returns a single row and returns this row
+
+        This opens a new connection and closes it once the query is done
+        """
+        with self.cur(**kwargs) as cur:
+            cur.execute(query, params=params)
+            result = cur.fetchall()
+
+            if allow_empty_result and len(result) == 0:
+                return None
+
+            assert len(result) == 1
```

Comment:
Instead of `assert len(result) == 1`, how about
`    raise Exception('sql_row returns more than one row ')`

Reply:
The nice thing about using an assert is that it shows what the ruselt is.